pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -667,10 +671,9 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, IControlled
             return 0;
         }
 
-        // As we update `_lastAumFeeCollectionTimestamp` when updating `_managementAumFeePercentage`, we only need to
-        // update `_lastAumFeeCollectionTimestamp` when non-zero AUM fees are paid. This avoids an SSTORE on zero-length
-        // collections.
-        _lastAumFeeCollectionTimestamp = block.timestamp;
+        // As we update the AUM fee collection timestamp when updating the AUM fee percentage, we only need to
+        // update the timestamp when non-zero AUM fees are paid. This avoids an SSTORE on zero-length collections.
+        _updateAumFeeCollectionTimestamp();
```

Comment:
`getManagementAumFeeParams()` reads `_aumState`, and here we read it again. Bit unfortunate that we need to do the warm sload, but that's likely better than passing state around, or having duplicate functions.

Reply:
uno_reverse_card.jpg